frontend/lib/screens/auth/login/login_screen.dart

```diff
@@ -40,45 +51,294 @@ class LoginScreen extends StatelessWidget {
-              LoginFormWidget(),
-              const SizedBox(
-                height: 45,
-              ),
-            ],
+          child: BlocConsumer<LoginBloc, LoginState>(
+            listener: (context, state) {
+              _blocListener(
+                  state, emailTextController, passwordTextController, context);
+            },
+            builder: (context, state) {
+              return _blocBuilder(loginBloc, state, emailTextController,
+                  passwordTextController, formKey, context);
+            },
           ),
         ),
       ),
     );
   }
 }
+
+void _blocListener(
+  LoginState state,
+  TextEditingController emailTextController,
+  TextEditingController passwordTextController,
+  BuildContext context,
+) {
+  if (state is LoginEdited) {
+    emailTextController.text = state.email;
+    passwordTextController.text = state.password;
+  }
+  if (state is LoginSuccess) {
+    AppRouter.goToMainScreen(context);
+  } else if (state is LoginFailure) {
+    ScaffoldMessenger.of(context).showSnackBar(
+      SnackBar(content: Text('Failed to log in: ${state.error.message}')),
+    );
+  }
+}
+
+Widget _blocBuilder(
+  LoginBloc loginBloc,
+  LoginState state,
+  TextEditingController emailTextController,
+  TextEditingController passwordTextController,
+  GlobalKey<FormState> formKey,
+  BuildContext context,
+) {
+  if (state is LoginLoading) {
+    return const CircularProgressIndicator();
+  } else {
+    return Form(
+      key: formKey,
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        mainAxisAlignment: MainAxisAlignment.start,
+        children: [
+          const SizedBox(
+            height: 45,
+          ),
+          CustomButtonComponent(
+            text: 'Log in with Google',
+            width: 328,
+            height: 48,
+            iconPath: 'assets/icons/google_icon.png',
+            onPressed: () {
+              loginBloc.add(LoginWithGoogleRequest());
+            },
+          ),
+          const SizedBox(
+            height: 45,
+          ),
+          Container(
+            padding: const EdgeInsets.symmetric(
+              horizontal: 40,
+            ),
+            child: const DividerWithTextComponent(
+              text: 'or log in with an email',
+              textSize: 24,
+            ),
+          ),
+          const SizedBox(
+            height: 45,
+          ),
+          FormTextfieldComponent(
+            fieldName: 'Email',
+            hintText: 'Enter your email',
+            obscureText: false,
+            controller: emailTextController,
+            onChanged: (value) {
+              loginBloc.add(EmailChanged(email: value));
+            },
+            validator: (value) {
+              if (value == null || value.isEmpty) {
+                return 'Please enter your email';
+              }
+
+              final emailRegex =
+                  RegExp(r'^[\w-]+(\.[\w-]+)*@[\w-]+(\.[\w-]+)+$');
```

Comment:
I'm wondering if we could move this regex to class'es private attribute. It could be initialized once, not every time of running this validator

Another idea is to create separe class `EmailValidator`, that will have method `validateEmail`

Reply:
Done in #23